src/Encoder.ts

```diff
@@ -389,7 +408,11 @@ export class Encoder<ContextType = undefined> {
       const value = object[key];
 
       if (!(this.ignoreUndefined && value === undefined)) {
-        this.encodeString(key);
+        if (typeof key === "string") {
+          this.encodeString(key);
+        } else {
+          this.encodeNumber(key);
```

Comment:
This will automatically encode as an appropriate number (admittedly, based on int mode when that PR lands) so there isn't a lot of control over the type - potentially we want that to be configurable e.g. to always set them to uint64 or similar?